src/shared/bus-polkit.c

```diff
@@ -588,3 +601,238 @@ Hashmap *bus_verify_polkit_async_registry_free(Hashmap *registry) {
         return hashmap_free(registry);
 #endif
 }
+
+static int varlink_check_good_user(Varlink *link, uid_t good_user) {
+        int r;
+
+        assert(link);
+
+        if (good_user == UID_INVALID)
+                return false;
+
+        uid_t peer_uid;
+        r = varlink_get_peer_uid(link, &peer_uid);
+        if (r < 0)
+                return r;
+
+        return good_user == peer_uid;
+}
+
+static int varlink_check_peer_privilege(Varlink *link) {
+        int r;
+
+        assert(link);
+
+        uid_t peer_uid;
+        r = varlink_get_peer_uid(link, &peer_uid);
+        if (r < 0)
+                return r;
+
+        uid_t our_uid = getuid();
+        return peer_uid == our_uid ||
+                (our_uid != 0 && peer_uid == 0);
+}
+
+#if ENABLE_POLKIT
+static int bus_message_new_polkit_auth_call_for_varlink(
+                sd_bus *bus,
+                Varlink *link,
+                const char *action,
+                const char **details,
+                bool interactive,
+                sd_bus_message **ret) {
+
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *c = NULL;
+        _cleanup_(pidref_done) PidRef pidref = PIDREF_NULL;
+        int r;
+
+        assert(bus);
+        assert(link);
+        assert(action);
+        assert(ret);
+
+        r = varlink_get_peer_pidref(link, &pidref);
+        if (r < 0)
+                return r;
+        if (r == 0) /* if we couldn't get a pidfd this returns == 0 */
+                return log_debug_errno(SYNTHETIC_ERRNO(EPERM), "Failed to get peer pidfd, cannot securely authenticate.");
+
+        uint64_t start_time;
+        r = pidref_get_start_time(&pidref, &start_time);
+        if (r < 0)
+                return r;
+
+        uid_t uid;
+        r = varlink_get_peer_uid(link, &uid);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_new_method_call(
+                        bus,
+                        &c,
+                        "org.freedesktop.PolicyKit1",
+                        "/org/freedesktop/PolicyKit1/Authority",
+                        "org.freedesktop.PolicyKit1.Authority",
+                        "CheckAuthorization");
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_append(
+                        c,
+                        "(sa{sv})s",
+                        "unix-process", 3,
+                        "pid", "u", (uint32_t) pidref.pid,
+                        "start-time", "t", (uint32_t) start_time,
+                        "uid", "u", (uint32_t) uid,
```

Comment:
`uid` is actually of type int32_t for this call (it's uint32 in different calls), so this cannot work and gets rejected https://gitlab.freedesktop.org/polkit/polkit/-/blob/123/src/polkit/polkitsubject.c?ref_type=tags#L433

Reply:
hm? this works perfectly here. the dialog pops up as it should, evertyhing works.

Reply:
ah, i guess it doesn't get reject, just ignored. what a stupid broken interface...

Reply:
yeah, it's really silly